src/frequenz/sdk/timeseries/battery_pool/_battery_pool_reference_store.py

```diff
@@ -0,0 +1,138 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""User interface for requesting aggregated battery-inverter data."""
+
+
+import asyncio
+import uuid
+from collections.abc import Awaitable, Set
+from datetime import timedelta
+from typing import Any
+
+from frequenz.channels import Receiver, Sender
+
+from ..._internal._asyncio import cancel_and_await
+from ...actor import ComponentMetricRequest, _channel_registry, _power_managing
+from ...actor.power_distributing._battery_pool_status import BatteryStatus
+from ...microgrid import connection_manager
+from ...microgrid.component import ComponentCategory
+from .._formula_engine import FormulaEnginePool
+from ._methods import MetricAggregator
+
+
+class BatteryPoolReferenceStore:  # pylint: disable=too-many-instance-attributes
+    """A class for maintaining the shared state/tasks for a set of pool of batteries.
+
+    This includes ownership of
+    - the formula engine pool and metric calculators.
+    - the tasks for updating the battery status for the metric calculators.
```

Comment:
Oh, this has `BackgroundService` written all over it.

Reply:
made an issue: https://github.com/frequenz-floss/frequenz-sdk-python/issues/715